src/components/BingoCard.tsx

```diff
@@ -1,31 +1,32 @@
-import React, { useState, useEffect, createRef } from "react";
+import React, { useState, useEffect, useRef } from "react";
 import "./BingoCard.css";
-import cards from "../assets/cards.json";
 import { useScreenshot, createFileName } from "use-react-screenshot";
 
 import { useSave } from "../hooks/useSave.ts";
 
-const BingoCard = ({ cardId }) => {
-  const [card, setCard] = useState([]);
-  const [marked, setMarked] = useState([]);
+export type Card = (number | null)[][];
+export type CardMarked = boolean[][];
+
+interface BingoCardProps {
+  cardIndex: number;
+  card: Card;
+}
+
+const BingoCard = ({ card, cardIndex }: BingoCardProps) => {
+  const [marked, setMarked] = useState<CardMarked>([]);
   const [isLoading, setIsLoading] = useState(true);
-  const ref = createRef(null);
+  const ref = useRef(null);
```

Comment:
Seems like `createRef()` is a more barebone version more applicable to class components. `useRef()` is more prevalent and moveing to ts unveiled that `createRef` doesnt accept any args.